src/Identity/Controllers/SsoController.cs

```diff
@@ -100,27 +108,28 @@ public async Task<IActionResult> Login(string returnUrl)
 
             return RedirectToAction(nameof(ExternalChallenge), new
             {
-                organizationIdentifier = domainHint,
+                domain_hint = domainHint,
                 returnUrl,
-                userIdentifier
+                userIdentifier,
+                ssoToken,
             });
         }
 
         [HttpGet]
-        public async Task<IActionResult> ExternalChallenge(string organizationIdentifier, string returnUrl,
-            string userIdentifier)
+        public async Task<IActionResult> ExternalChallenge(string domain_hint, string returnUrl,
```

Comment:
Code Style: Parameters should be camelCase and should not have underscores.
```suggestion
        public async Task<IActionResult> ExternalChallenge(string domainHint, string returnUrl,
```